Reset password via email, user profile edit, admin settings

pellecchialuigi opened the pull request:
This PR solves the #60 

Added admin settings page to configure the smtp server and the APP endpoint.
Added frontend and backend support to:
- Reset password via email
- Edit user profile information such as username and password

The password reset works as following:
in the users table I added a field for a temporary password and a field for a token.
The temporary password is shared via email to the user and a link with the token is provided as part of the email to activate the new password. If the email is not wanted by the user nothing happen as the temporary password is not active. So user can continue to login with his original password.

If the reset is wanted, user can press the link in the email and temporary password will be activated.
If admin configure the APP endpoint correctly user will be redirect to the login page of the frontend.

@nopeppermint May I ask your review on this PR?

nopeppermint commented:
strange, I am no longer able to login as admin with the default password 1234, does not happen with main branch.

I just checked out this branch and run the ./run_demo.sh script

I get "Error: TypeError: NetworkError when attempting to fetch resource."

pellecchialuigi commented:
Good catch, I'm sorry I made a mistake, I changed the default value of the API Endpoint.
I just fixed it. May I ask you to give it another try?

I want to point out that if you deploy BASIL on top of an existing deployment, in other words if you keep your existing podman volumes, you should run a database migration script to update from last db version 1_5_0 to the version of this PR 1_6_3. The script is under db/models/migration.
That because I added fields to the users table.

To do that you can inspect your volume to see the location of your basil.db file and then you can run

sqlite3 location/of/my/db/basil.db < db/models/migration/sqlite_1_6_3.sql

nopeppermint commented:
@pellecchialuigi I stopped using BASIL will also stop my efforts in helping the development of BASIL, sorry.

pellecchialuigi commented:
Thanks for letting me know and for all your good inputs. See you
